asapdiscovery-data/asapdiscovery/data/schema_v2/ligand.py

```diff
@@ -0,0 +1,151 @@
+from pathlib import Path
+from typing import Any, Dict, Optional, Tuple, Union  # noqa: F401
+from uuid import UUID
+
+from asapdiscovery.data.openeye import (
+    get_SD_data,
+    get_SD_data_dict,
+    oechem,
+    oemol_to_inchi,
+    oemol_to_inchikey,
+    oemol_to_sdf_string,
+    oemol_to_smiles,
+    print_SD_Data,
+    sdf_string_to_oemol,
+    set_SD_data,
+    set_SD_data_dict,
+    smiles_to_oemol,
+)
+from asapdiscovery.data.schema import ExperimentalCompoundData
+from pydantic import UUID4, Field
+
+from .schema_base import (
+    DataModelAbstractBase,
+    DataStorageType,
+    read_file_directly,
+    write_file_directly,
+)
+
+
+class InvalidLigandError(ValueError):
+    ...
+
+
+# Ligand Schema
+
+
+class LigandIdentifiers(DataModelAbstractBase):
+    """
+    Identifiers for a Ligand
+    """
+
+    moonshot_compound_id: Optional[str] = Field(
+        None, description="Moonshot compound ID"
+    )
+    manifold_api_id: Optional[UUID] = Field(
+        None, description="Unique ID from Postera Manifold API"
+    )
+    manifold_vc_id: Optional[str] = Field(
+        None, description="Unique VC ID (virtual compound ID) from Postera Manifold"
+    )
+    compchem_id: Optional[UUID4] = Field(
+        None, description="Unique ID for P5 compchem reference"
+    )
+
+
+class Ligand(DataModelAbstractBase):
+    """
+    Schema for a Ligand
+    """
+
+    compound_name: str = Field(None, description="Name of compound")
+    ids: Optional[LigandIdentifiers] = Field(
+        None,
+        description="LigandIdentifiers Schema for identifiers associated with this ligand",
+    )
+    experimental_data: Optional[ExperimentalCompoundData] = Field(
+        None,
+        description="ExperimentalCompoundData Schema for experimental data associated with the compound",
+    )
+    data: str = Field(
+        "",
+        description="SDF file stored as a string to hold internal data state",
+        repr=False,
+    )
+    data_format: DataStorageType = Field(
+        DataStorageType.sdf,
+        description="Enum describing the data storage method",
+        allow_mutation=False,
+    )
```

Comment:
is there a reason these can't go into the baseclass?

Reply:
Yes because it is set to `DataStorageType.sdf` as a constant in this subclass, with `allow_mutation=False`. `Targets`  in #339 use the other enum member `DataStorageType.pdb` immutably in the same way. :)

Reply:
Oh, right of course! Sorry late night brain. Does it make sense to make these two methods abstractmethods in the base class though?

Reply:
ah not really because it isn't functional, don't think it adds anything moving to baseclass as would need to re-define to get correct description.

Reply:
if the `Ligand` object will only ever have an SDF file in the `data` object, would it make sense/is it possible to just make the `data_format` a constant rather than a `Field`?

Reply:
The field should be const now with `allow_mutation=False`, `const=True`